src/com/owncloud/android/ui/dialog/RateMeDialog.java

```diff
@@ -0,0 +1,165 @@
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.owncloud.android.ui.dialog;
+
+import android.app.Dialog;
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.Button;
+
+import com.owncloud.android.R;
+
+public class RateMeDialog extends DialogFragment {
+
+    private Dialog dialog;
+
+    private static final String ARG_CANCELABLE = RateMeDialog.class.getCanonicalName() + ".ARG_CANCELABLE";
+    private static final String APP_PACKAGE_NAME = RateMeDialog.class.getCanonicalName() + ".APP_PACKAGE_NAME";
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+        setCancelable(false);
+    }
+
+    /**
+     * Public factory method to get dialog instances.
+     *
+     * @param packageName   The package name of the application
+     * @param cancelable    If 'true', the dialog can be cancelled by the user input (BACK button, touch outside...)
+     * @return              New dialog instance, ready to show.
+     */
+    public static RateMeDialog newInstance(String packageName, boolean cancelable) {
+        RateMeDialog fragment = new RateMeDialog();
+        Bundle args = new Bundle();
+        args.putBoolean(ARG_CANCELABLE, cancelable);
+        args.putString(APP_PACKAGE_NAME, packageName);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Create a view by inflating desired layout
+        View v = inflater.inflate(R.layout.rate_me_dialog, container,  false);
+
+        Button rateNowButton = (Button) v.findViewById(R.id.button_rate_now);
+        Button laterButton = (Button) v.findViewById(R.id.button_later);
+        Button noThanksButton = (Button) v.findViewById(R.id.button_no_thanks);
+
+        rateNowButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String packageName = getArguments().getString(APP_PACKAGE_NAME);
+                Uri uri = Uri.parse("market://details?id=" + packageName);
```

Comment:
I would create a new constant in the top of the class to include `"market://details?id="`